overlord/ifacestate/handlers.go

```diff
@@ -227,6 +227,24 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// connect OS snap if snap name is omitted
+	if slotRef.Snap == "" {
+		slotRef.Snap = "ubuntu-core"
```

Comment:
Please sync this with Chipaca so that we don't make the problem more complicated.